### PR TITLE
Ameliorate Warnings, PDN Generation, and the default design

### PR DESCRIPTION
**Pull request by donn**
~ Reimplemented the default serial/parallel multiplier to be a bit less cryptic and eliminate linter warnings; and added a proper testbench
~ Reimplemented deprecated variable translation behavior to account for the situation where a deprecated variable's value does not match the default for the new value, which causes a crash
~ Slightly improved warning for designs having been black-boxed during STA
~ PDN Generation Updates
    ~ Renamed `DESIGN_IS_CORE` to `FP_PDN_MULTILAYER` with translation behavior
    ~ PDN generation will now always attempt to extend to core ring (which is inert if core rings do not exist)
    ~ Fixed bug where `FP_PDN_MULTILAYER` being set to `0` would attempt to create a core-ring on two layers anyway
~ IR drop now prints a warning if `VSRC_LOC_FILE` is not provided
- Removed deprecation behavior for `GLB_RT` variables - it's been over a year (>=6 mo as promised)

---

Depends on https://github.com/efabless/openlane-ci-designs/pull/5

### CODE REVIEW

File: scripts/openroad/common/pdn_cfg.tcl [hunk 1]
```diff
@@ -108,12 +104,21 @@ if { $::env(FP_PDN_ENABLE_RAILS) == 1 } {
...
+        add_pdn_ring \
+            -grid stdcell_grid \
+            -layers "$::env(FP_PDN_VERTICAL_LAYER) $::env(FP_PDN_HORIZONTAL_LAYER)" \
+            -widths "$::env(FP_PDN_CORE_RING_VWIDTH) $::env(FP_PDN_CORE_RING_HWIDTH)" \
+            -spacings "$::env(FP_PDN_CORE_RING_VSPACING) $::env(FP_PDN_CORE_RING_HSPACING)" \
+            -core_offset "$::env(FP_PDN_CORE_RING_VOFFSET) $::env(FP_PDN_CORE_RING_HOFFSET)"
+    } else {
+        add_pdn_ring \
+            -grid stdcell_grid \
+            -layers "$::env(FP_PDN_VERTICAL_LAYER)" \
```
Comment (kareefardi) [inline scripts/openroad/common/pdn_cfg.tcl:117]: Getting this error when entering the else statement
[ERROR PDN-1012] Expecting a list of 2 elements for -layers option of add_pdn_ring command, found 1.
Also doesn't make sense for a ring to have only one layer.

File: scripts/tcl_commands/sta.tcl [hunk 2]
```diff
@@ -79,11 +79,15 @@ proc run_sta {args} {
...
         set file_path [read $fp]
+        set modules [list]
         foreach line [split $file_path "\n"] {
             if { [regexp {module\s+(\S+)\s+not\s+found} $line match first_group] } {
-                puts_warn "Module $first_group blackboxed during sta"
+                lappend modules $first_group
             }
         }
+        if { [llength $modules] > 0 } {
+            puts_warn "The following modules were black-boxed for STA as there was no timing information found: \[[join $modules ", "]\]"
```
Comment (kareefardi) [inline scripts/tcl_commands/sta.tcl:89]: I would prefer replacing ` \[[join $modules ", "]\]` with `\n \[[join $modules "\n\t"]\]`.

File: scripts/tcl_commands/all.tcl [hunk 3]
```diff
@@ -616,64 +652,6 @@ proc prep {args} {
...
-    ### GLB_RT -> GRT (Document using ‡)
-    handle_deprecated_config GLB_RT_ALLOW_CONGESTION GRT_ALLOW_CONGESTION
-    handle_deprecated_config GLB_RT_OVERFLOW_ITERS GRT_OVERFLOW_ITERS
-    handle_deprecated_config GLB_RT_ANT_ITERS GRT_ANT_ITERS
-    handle_deprecated_config GLB_RT_ESTIMATE_PARASITICS GRT_ESTIMATE_PARASITICS
-    handle_deprecated_config GLB_RT_MAX_DIODE_INS_ITERS GRT_MAX_DIODE_INS_ITERS
-    handle_deprecated_config GLB_RT_OBS GRT_OBS
-    handle_deprecated_config GLB_RT_ADJUSTMENT GRT_ADJUSTMENT
-    handle_deprecated_config GLB_RT_MACRO_EXTENSION GRT_MACRO_EXTENSION
-    handle_deprecated_config GLB_RT_LAYER_ADJUSTMENTS GRT_LAYER_ADJUSTMENTS
```
Comment (kareefardi) [inline scripts/tcl_commands/all.tcl:671]: did you deprecate some variables? i.e. remove `handle_deprecated_config` for some variables.

**Comment by kareefardi** [comment]
Can you also do same as https://github.com/efabless/openlane2-ci-designs/pull/14 here. Should fix https://github.com/The-OpenROAD-Project/OpenLane/issues/1978

File: scripts/tcl_commands/all.tcl (hunk 3 above)
Comment (donn) [reply scripts/tcl_commands/all.tcl:671]: Yes-- our written policy is >=6 months and the `GLB_RT` variables have been around for about a year, give or take.

File: scripts/openroad/common/pdn_cfg.tcl (hunk 1 above)
Comment (donn) [reply scripts/openroad/common/pdn_cfg.tcl:117]: I added an error.

File: scripts/tcl_commands/sta.tcl (hunk 2 above)
Comment (donn) [reply scripts/tcl_commands/sta.tcl:98]: Did something similar-ish.

**Review by kareefardi — DISMISSED** [review]

**Pull request merged**